### PR TITLE
Remove unnecessary null check in DomainNameHelper

### PR DESCRIPTION
**Pull request by stephentoub**
Found by a coverity scan

cc: @cipop, @davidsh

### CODE REVIEW

**Comment by stephentoub** [comment]
Test Innerloop Windows_NT Release Build and Test please (known AV in runtime)

File: src/System.Private.Uri/src/System/DomainNameHelper.cs [hunk 1]
```diff
@@ -324,7 +324,7 @@ internal unsafe static string IdnEquivalent(char* hostname, int start, int end,
             {
                 // just lowercase for ascii
                 string unescapedHostname = new string(hostname, start, end - start);
-                return ((unescapedHostname != null) ? unescapedHostname.ToLowerInvariant() : null);
+                return unescapedHostname.ToLowerInvariant();
```
Comment (davidsh) [inline src/System.Private.Uri/src/System/DomainNameHelper.cs]: I was going to suggest that we should have an Assert about not being null.  But maybe the "coverity" tool (I'm not that familiar with it) is sufficient for static analysis.

**Comment by davidsh** [comment]
LGTM

File: src/System.Private.Uri/src/System/DomainNameHelper.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Private.Uri/src/System/DomainNameHelper.cs]: A constructor will never return null.  If it did, that would be a giant CLR bug.

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu14.04 Debug Build and Test please

**Pull request merged**